pkg/resource/internal/compare/compare.go

```diff
@@ -0,0 +1,108 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package compare implements term operation helpers.
+package compare
+
+import (
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// GetNumbers returns numbers parsed from left and right params, if any of string to number conversion fails returns false as the 3rd arg.
+func GetNumbers(left, right string) (int64, int64, bool) {
+	numLeft, ok := parseValue(left)
+	if !ok {
+		return 0, 0, false
+	}
+
+	numRight, ok := parseValue(right)
+	if !ok {
+		return 0, 0, false
+	}
+
+	return numLeft, numRight, true
+}
+
+func parseValue(value string) (int64, bool) {
+	value = strings.TrimSpace(value)
+
+	var (
+		digits string
+		rest   string
+	)
+
+	for _, letter := range value {
+		if rest == "" && unicode.IsDigit(letter) || string(letter) == "." {
+			digits += string(letter)
+
+			continue
+		}
+
+		rest += string(letter)
+	}
+
+	if len(digits) == 0 {
+		return 0, false
+	}
+
+	res, err := strconv.ParseInt(digits, 10, 64)
+	if err != nil {
+		return 0, false
+	}
+
+	return res * getMultiplier(rest), true
+}
+
+func getMultiplierForUnits(value string) (int64, bool) {
+	var val int64
+
+	switch strings.ToLower(value) {
+	case "pi":
+		val = 1 << 50
+	case "ti":
+		val = 1 << 40
+	case "gi":
+		val = 1 << 30
+	case "mi":
+		val = 1 << 20
+	case "ki":
+		val = 1 << 10
+	case "p":
+		val = 1e15
+	case "t":
+		val = 1e12
+	case "g":
+		val = 1e9
+	case "m":
+		val = 1e6
+	case "k":
+		val = 1e3
+	default:
+		return 0, false
+	}
+
+	return val, true
+}
+
+func getMultiplier(value string) int64 {
```

Comment:
I think this can be simplified a bit by simply doing a match on `value[:2]` if the len > 2, and then for `value[:1]` if the previous one never matched? 

as the function `getMultiplierForUnits` is already explicitly matching for either one-character or two-character prefix?